orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/graph/transform/ConfigStageInjectionTransform.java

```diff
@@ -75,17 +72,107 @@ private void replaceStages(PipelineTemplate pipelineTemplate) {
   }
 
   private void injectStages(PipelineTemplate pipelineTemplate) {
-    List<StageDefinition> configStages = templateConfiguration.getStages()
+    // Create initial graph via dependsOn.
+    createDag(pipelineTemplate.getStages());
+
+    // Handle stage injections.
+    injectStages(pipelineTemplate.getStages(), pipelineTemplate.getStages());
+    injectStages(templateConfiguration.getStages(), pipelineTemplate.getStages());
+  }
+
+  private static void createDag(List<StageDefinition> stages) {
+    stages
```

Comment:
Do we have a sensible way to detect cycles ? Or is the assumption that the user will never introduce a cycle.
I would be inclined to do a first pass through the iteration to make sure we aren't going to get hit with a stack overflow. Thoughts ?